src/tx/penalty/transaction.ts

```diff
@@ -86,40 +86,20 @@ export async function injectPenaltyTX(
   // store the unsignedTx to local map for later use
   recordPenaltyTX(txId, signedTx)
 
-  // inject the unsignedTx if we are lucky node
-  const consensusNodes = shardus.getConsenusGroupForAccount(signedTx.reportedNodePublickKey)
-  const consensusNodeIdsAndRanks: { nodeId: string; rank: bigint }[] = consensusNodes.map((n) => {
-    return {
-      nodeId: n.id,
-      rank: shardus.computeNodeRank(n.id, txId, signedTx.timestamp),
-    }
-  })
-  const sortedNodeIdsAndRanks = consensusNodeIdsAndRanks.sort((a, b): number => {
-    return b.rank > a.rank ? 1 : -1
-  })
-  const ourNodeId = shardus.getNodeId()
-
-  let isLuckyNode = false
-  for (let i = 0; i < ShardeumFlags.numberOfNodesToInjectPenaltyTx; i++) {
-    if (sortedNodeIdsAndRanks[i].nodeId === ourNodeId) {
-      isLuckyNode = true
-      break
+  // Limit the nodes that send this to the 5 closest to the node id
+  const closestNodes = shardus.getClosestNodes(eventData.nodeId, 5)
```

Comment:
This has to be node.publicKey

Reply:
Also, I think we should use ShardeumFlags.numberOfNodesToInjectPenaltyTx instead of hard-coded value 5